languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt

```diff
@@ -155,25 +155,25 @@ class JavaHttpRequestRenderer @Inject constructor(override val vrapTypeProvider:
                 .map { "this.$it" }
                 .joinToString(separator = ", ")
 
-        val bodyName: String? = if (this.bodies != null && this.bodies.isNotEmpty()) {
-            if (this.bodies[0].type.toVrapType() is VrapObjectType) {
+        val bodyName : String? = if(this.bodies != null && this.bodies.isNotEmpty()){
+            if(this.bodies[0].type.toVrapType() is VrapObjectType) {
                 val methodBodyVrapType = this.bodies[0].type.toVrapType() as VrapObjectType
                 methodBodyVrapType.simpleClassName.decapitalize()
-            } else {
+            }else {
                 "jsonNode"
             }
-        } else {
+        }else {
             null
         }
-
-        val addingQueryParams: String = this.queryParameters
-                .filter { it.annotations.find { it.type.name.equals("placeholderParam") } == null }
-                .map { "params.add(this.${it.fieldName()}.stream().map(s -> \"${it.name}=\" + ${if (it.type.name.equals("string")) "urlEncode(s)" else "s"}).collect(Collectors.joining(\"&\")));" }
+        
+        val addingQueryParams : String = this.queryParameters
+                .filter { it.annotations.find { it.type.name.equals("placeholderParam") } == null}
```

Comment:
Is this the same as the constant `PLACEHOLDER_PARAM_ANNOTATION`? How about extracting the code to a separate method with a name that reveals the intention of `it.annotations.find { it.type.name.equals("placeholderParam")`?

Reply:
I also dont know the intention, it is there just to ignore query params with that annotation.

Reply:
See https://github.com/commercetools/commercetools-api-reference/blob/master/resources/product-projections-suggest.raml#L14

with the help of this annotation is possible to generate methods with placeholder values like https://github.com/commercetools/commercetools-php-raml/blob/master/lib/commercetools-api/test/unit/Client/Resource/ResourceByProjectKeyProductProjectionsSuggestTest.php#L50 and https://github.com/commercetools/commercetools-php-raml/blob/4fa6b807cf9e3c06454c3e4dce67e76c472a2f46/lib/commercetools-api/src/Client/Resource/ByProjectKeyProductProjectionsSuggestGet.php#L141

The filter just avoids to implement this behavior